### PR TITLE
[MIG] membership_prorate_variable_period: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
Helper module for making the prorate correctly when you are dealing with
variable periods in your memberships, which functionality is provided by
the module *membership_variable_period*.

Depends on:

- [x] https://github.com/OCA/vertical-association/pull/70 `membership_prorate`
- [x] https://github.com/OCA/vertical-association/pull/69 `membership_variable_period`

cc @Tecnativa

### CODE REVIEW

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1: 

Please , how to set maturty to high? :smile: this is the 5th version with this modules

File: membership_prorate_variable_period/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @api.multi
+    def _get_next_date(self, date, qty=1):
+        res = super(ProductTemplate, self)._get_next_date(date)
+        if self.membership_interval_unit == 'days':
+            raise exceptions.Warning(
+                _("It's not possible to prorate daily periods."))
+        qty = int(math.ceil(qty)) * self.membership_interval_qty
+        if self.membership_interval_unit == 'weeks':
+            weekday = date.weekday()
+            date_from = date - datetime.timedelta(weekday)
```
Comment (Tardo) [inline membership_prorate_variable_period/models/product_template.py]: Perhaps can do it more 'Odooish' using 'date_utils'

Comment (chienandalu) [reply membership_prorate_variable_period/models/product_template.py]: Thanks @Tardo I refactored it using date_utils as you suggested ;)

**Comment by pedrobaeza** [comment]
@Tardo please confirm everything is alright

File: membership_prorate_variable_period/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def _get_next_date(self, date, qty=1):
+        next_date = super(ProductTemplate, self)._get_next_date(date)
+        if self.membership_interval_unit == 'days':
+            raise exceptions.Warning(
+                _("It's not possible to prorate daily periods."))
+        qty = int(math.ceil(qty)) * self.membership_interval_qty
```
Comment (Tardo) [inline membership_prorate_variable_period/models/product_template.py]: math.ceil return an integer, cast it is redundant.

**Review by Tardo — APPROVED** [review]
LGTM, code review.

File: membership_prorate_variable_period/models/product_template.py (hunk 2 above)
Comment (chienandalu) [reply membership_prorate_variable_period/models/product_template.py]: Sure :) changed

File: membership_prorate_variable_period/README.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+~~~~~~~
+
+* Tecnativa
+
+Contributors
+~~~~~~~~~~~~
+
+* Pedro M. Baeza <pedro.baeza@tecnativa.com>
+* David Vidal <david.vidal@tecnativa.com>
+
```
Comment (rafaelbn) [inline membership_prorate_variable_period/README.rst:85]: I was there and I'm here year by year, from v8...

**Pull request merged**